### PR TITLE
Add metric for num decompressed bytes that pass through zstd encoders/decoders

### PR DESCRIPTION
**Pull request by maggie-lou**
---

**Version bump**: TODO

### CODE REVIEW

**Comment by maggie-lou** [comment]
Current dependencies on/for this PR:
* master
  * **PR #2956** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/2956" target="_blank">Graphite</a> 
    * **PR #2838** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/2838" target="_blank">Graphite</a> 
      * **PR #2932** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/2932" target="_blank">Graphite</a> 
      * **PR #2962** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/2962" target="_blank">Graphite</a> 
        * **PR #2966** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/2966" target="_blank">Graphite</a> 
          * **PR #2967** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/2967" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/2967?utm_source=stack-comment).

File: server/metrics/metrics.go [hunk 1]
```diff
@@ -1418,6 +1421,15 @@ var (
 		ClickhouseStatusLabel,
 	})
 
+	BytesCompressed = promauto.NewCounterVec(prometheus.CounterOpts{
```
Comment (bduffany) [inline server/metrics/metrics.go:1424]: nit: I think it'd be clearer to have a separate metric for `BytesDecompressed`

File: server/util/compression/compression.go [hunk 2]
```diff
@@ -163,8 +169,9 @@ func NewZstdDecompressingReader(reader io.Reader) (io.ReadCloser, error) {
 	pr, pw := io.Pipe()
 	go func() {
 		// Write decoded bytes to pw and pipe to pr
-		_, err = decoder.WriteTo(pw)
+		n, err := decoder.WriteTo(pw)
 		pw.CloseWithError(err)
+		metrics.BytesCompressed.With(prometheus.Labels{metrics.CacheRequestType: "zstdDecompress"}).Add(float64(n))
```
Comment (bduffany) [inline server/util/compression/compression.go:174]:
```suggestion
		metrics.BytesCompressed.With(prometheus.Labels{metrics.CompressionType: "zstd"}).Add(float64(n))
```

File: server/metrics/metrics.go [hunk 3]
```diff
@@ -1418,6 +1421,15 @@ var (
...
+		Namespace: bbNamespace,
+		Subsystem: "compressor",
+		Name:      "bytes_compressed",
+		Help:      "Num decompressed bytes passed through compressors",
```
Comment (bduffany) [inline server/metrics/metrics.go:1428]:
```suggestion
		Help:      "The number of decompressed bytes passed through compressors.",
```

Comment (maggie-lou) [reply server/metrics/metrics.go]: Done

File: server/util/compression/compression.go [hunk 4]
```diff
@@ -81,7 +86,8 @@ func NewZstdDecompressor(writer io.Writer) (io.WriteCloser, error) {
 			}
 		}()
 		defer pr.Close()
-		_, err := decoder.WriteTo(writer)
+		n, err := decoder.WriteTo(writer)
+		metrics.BytesCompressed.With(prometheus.Labels{metrics.CacheRequestType: "zstdDecompress"}).Add(float64(n))
```
[the remark this answers is not in the document]
Comment (maggie-lou) [reply server/util/compression/compression.go]: Oops - fixed

File: server/metrics/metrics.go (hunk 1 above)
Comment (maggie-lou) [reply server/metrics/metrics.go:1424]: Done

**Review by bduffany — APPROVED** [review]
nice!

**Pull request merged**